modin/pandas/test/test_concat.py

```diff
@@ -236,3 +242,58 @@ def test_sort_order(sort, join, axis):
         modin_concat,
     )
     assert list(pandas_concat.columns) == list(modin_concat.columns)
+
+
+@pytest.mark.skipif(
+    StorageFormat.get() not in ("Hdk", "Base"),
```

Comment:
Looks like, your changes in `general.py` do not fix the problem on the pandas backend?

Reply:
Right. On the master branch, the test fails on all backends. On this branch, the test passes on HDK and Base.

Reply:
Do you have a separate reproducer for the pandas backend where the previous filtering implementation was not working properly? It is highly desirable to add a test for this in order to have protection in the future.

Reply:
I've filed a bug with a reproducer - https://github.com/modin-project/modin/issues/5696

Reply:
You describe another problem in the title. Isn't it?

Reply:
The test, in the issues description, reproduces multiple problems and this is just one of them.
Changed the issue title.